judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java

```diff
@@ -417,17 +417,131 @@ void testEntityCollectionQueriesDao() {
                 containsInAnyOrder(d.identifier())
         );
 
-        // With parameter and limit
-        assertEquals(
-                1,
-                entityCollectionWithValueParamQueryDao
+        // With parameter, limit and offset
+        List<EntityQueryElement> list = entityCollectionWithValueParamQueryDao
+                .query(EntityCollectionWithValueParamQueryParameter
+                        .builder()
+                        .withValue(4)
+                        .build())
+                .orderByDescending(EntityQueryElementAttribute.NAME)
+                .execute(1);
+
+        assertEquals(1, list.size());
+        assertEquals(e.identifier(), list.get(0).identifier());
+
+        list = entityCollectionWithValueParamQueryDao
+                .query(EntityCollectionWithValueParamQueryParameter
+                        .builder()
+                        .withValue(4)
+                        .build())
+                .orderByDescending(EntityQueryElementAttribute.NAME)
+                .execute(null);
+
+        assertEquals(2, list.size());
+        assertEquals(e.identifier(), list.get(0).identifier());
+        assertEquals(d.identifier(), list.get(1).identifier());
```

Comment:
This is very risky considering that the order of the IDs of 'e' and 'd' could be different; this test fails, but it should not.

The following example could be used for each check if the size is known:

```
assertEqual(1, list.stream().filter(i -> i.identifier().equals(e.identifier())).count());
```

This request also applies to the following occurrences.

Reply:
sry, didn't check query details. order is indeed always the same